src/jit/codegencommon.cpp

```diff
@@ -6507,6 +6507,8 @@ void CodeGen::genSetGSSecurityCookie(regNumber initReg, bool* pInitRegZeroed)
         //  mov   dword ptr [frame.GSSecurityCookie], #GlobalSecurityCookieVal
         instGen_Store_Imm_Into_Lcl(TYP_I_IMPL, EA_PTRSIZE, compiler->gsGlobalSecurityCookieVal,
                                    compiler->lvaGSSecurityCookie, 0, initReg);
+
+        *pInitRegZeroed = false;
```

Comment:
We now have `*pInitRegZeroed = false` both in the `if` and the `else`. Can we set it once outside of the if/else?

Reply:
It's not set for `_TARGET_XARCH_` in the `if`. Also, it should not be set for `_TARGET_XARCH_` in the `else`.

Reply:
This code is a bit difficult to reason about. Is it possible that `initReg` can be `REG_EAX` on `_TARGET_XARCH_`? If so, would we then need to set `*pInitRegZeroed = false`?
It would be great to add more comments to clarify - potentially in a new function header?

Reply:
This function is indeed difficult to reason and it has one issue that I am fixing in this PR and the conservative assignment `*pInitRegZeroed = false` that Bruce pointed out.

I am looking into splitting the function (or at least if-else block) into `_TARGET_ARMARCH_` and `_TARGET_XARCH_` and see if this helps to improve readability. 

Although, my original plan was to deliver correctness fix in one PR and follow-up later with the refactoring changes now I am thinking about doing both as a part of one PR but in separate commits.